File: panda-ij-plugin/src/eu/nyerel/panda/ijplugin/runner/PandaRunConfigurationExtension.java
package eu.nyerel.panda.ijplugin.runner;

import com.intellij.execution.ExecutionException;
import com.intellij.execution.RunConfigurationExtension;
import com.intellij.execution.application.ApplicationConfiguration;
import com.intellij.execution.configurations.JavaParameters;
import com.intellij.execution.configurations.ParametersList;
import com.intellij.execution.configurations.RunConfigurationBase;
import com.intellij.execution.configurations.RunnerSettings;
import com.intellij.openapi.options.SettingsEditor;
import com.intellij.openapi.util.InvalidDataException;
import com.intellij.openapi.util.WriteExternalException;
import com.intellij.ui.classFilter.ClassFilter;
import org.jdom.Element;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.io.File;
import java.net.URISyntaxException;
import java.net.URL;

/**
 * @author Rastislav Papp (rastislav.papp@gmail.com)
 */
public class PandaRunConfigurationExtension extends RunConfigurationExtension {

	@Nullable
	@Override
	protected <P extends RunConfigurationBase> SettingsEditor<P> createEditor(@NotNull P configuration) {
		return new PandaSettingsEditor<P>(configuration.getProject());
	}

	@Override
	public <P extends RunConfigurationBase> void updateJavaParameters(P configuration, JavaParameters params, RunnerSettings runnerSettings) throws ExecutionException {
	}

	@Override
	protected void readExternal(@NotNull RunConfigurationBase runConfiguration, @NotNull Element element)
			throws InvalidDataException {

	}

	@Override
	protected void writeExternal(@NotNull RunConfigurationBase runConfiguration, @NotNull Element element)
			throws WriteExternalException {

	}

	@Nullable
	@Override
	protected String getEditorTitle() {
		return "Panda";
	}

	@Override
	protected boolean isApplicableFor(@NotNull RunConfigurationBase configuration) {
		return true;
	}

}

File: panda-ij-plugin/src/eu/nyerel/panda/ijplugin/runner/PandaRunConfigurationExtension.kt
package eu.nyerel.panda.ijplugin.runner

import com.intellij.execution.ExecutionException
import com.intellij.execution.RunConfigurationExtension
import com.intellij.execution.configurations.JavaParameters
import com.intellij.execution.configurations.RunConfigurationBase
import com.intellij.execution.configurations.RunnerSettings
import com.intellij.openapi.options.SettingsEditor
import com.intellij.openapi.util.InvalidDataException
import com.intellij.openapi.util.WriteExternalException
import org.jdom.Element

/**
 * @author Rastislav Papp (rastislav.papp@gmail.com)
 */
class PandaRunConfigurationExtension : RunConfigurationExtension() {

    override fun <P : RunConfigurationBase> createEditor(configuration: P): SettingsEditor<P>? {
        return PandaSettingsEditor(configuration.project)
    }

    @Throws(ExecutionException::class)
    override fun <P : RunConfigurationBase> updateJavaParameters(configuration: P, params: JavaParameters, runnerSettings: RunnerSettings?) {
    }

    @Throws(InvalidDataException::class)
    override fun readExternal(runConfiguration: RunConfigurationBase, element: Element) {

    }

    @Throws(WriteExternalException::class)
    override fun writeExternal(runConfiguration: RunConfigurationBase, element: Element) {

    }

    override fun getEditorTitle(): String? {
        return "Panda"
    }

    override fun isApplicableFor(configuration: RunConfigurationBase): Boolean {
        return true
    }

}
